Add specifying edge color with RGBA array

jbcdnr opened the pull request.

mdeff commented on pygsp/plotting.py:
```diff
@@ -318,7 +318,9 @@ def _plot_graph(G, vertex_color, vertex_size, highlight,
         ranges from 0.2 to 0.9.
         If None, edge color is set to `graph.plotting['edge_color']`.
-        Alternatively, a color can be set in any format accepted by matplotlib.
+        Alternatively, a color can be set in any format accepted by matplotlib,
```
```suggestion
        Alternatively, a color can be set in any format accepted by matplotlib.
```

mdeff commented on pygsp/plotting.py:
```diff
@@ -318,7 +318,9 @@ def _plot_graph(G, vertex_color, vertex_size, highlight,
-        Alternatively, a color can be set in any format accepted by matplotlib.
+        Alternatively, a color can be set in any format accepted by matplotlib,
+        Each edge color can by specified by an RGB(A) array of dimension
+        `n_edges` x 3 (or 4).
```
Add the same description for node color.

jbcdnr commented:
@mdeff I updated the doc.

Also I fixed the default behaviour for 1D signal: should be considered as a signal and not a grey scale by Matplotlib. Let me know if you see a cleaner way to check this condition than `try` `except` (bede579e357df9dafa7e760547a766d152f24546).

mdeff commented:
What's the issue with 1D signals? I've tested with 1D signals, and all signals that are not of length 3 or 4 (where there's an unsolvable ambiguity) aren't considered colors. Single numbers are not considered color either.

```
>>> mpl.colors.is_color_like(0.8)
False
>>> mpl.colors.is_color_like((0.8,)*1)
False
>>> mpl.colors.is_color_like((0.8,)*2)
False
>>> mpl.colors.is_color_like((0.8,)*3)
True
>>> mpl.colors.is_color_like((0.8,)*4)
True
>>> mpl.colors.is_color_like((0.8,)*5)
False
>>> mpl.colors.is_color_like((0.8,)*10)
False
```

Also, no test failed without that commit.

jbcdnr commented:
```
>>> mpl.colors.is_color_like((0.8,)*4)
True
```
1D signal of values in [0, 1] of length 3 or 4 would be considered as grey scale and would not be normalised as expected (> 0.25). The bug is also present in `master` and would be fixed with this. However the control flow is not very elegant.

mdeff commented:
I see. That's indeed the unsolvable ambiguity. It happens with  `pg.graphs.Sensor(3, k=1).plot([1, 0, 0])`. Matplotlib warns about the situation if we run `pg.graphs.Sensor().plot([1, 0, 0])`.

```
'c' argument looks like a single numeric RGB or RGBA sequence, which should be
avoided as value-mapping will have precedence in case its length matches with 'x' & 'y'. 
Please use a 2-D array with a single row if you really want to specify the same RGB
or RGBA value for all points.
```

I think that the corner case it covers (graphs with 3 or 4 nodes or edges) is not worth the complexity. Moreover, the case can be resolved on the user side by following matplotlib's advice.
* ambiguous (wrongly plotted)
  ```
   >>> color = np.array([1, 0, 0])
   >>> mpl.colors.is_color_like(color)
   True
   >>> all(map(mpl.colors.is_color_like, color))
   False
   >>> pg.graphs.Sensor(3, k=1).plot(color)
   ```
* single color as row vector (correctly plotted)
  ```
   >>> color = np.array([[1, 0, 0]])
   >>> mpl.colors.is_color_like(color)
   False
   >>> all(map(mpl.colors.is_color_like, color))
   True
   >>> pg.graphs.Sensor(3, k=1).plot(color)
   ```
* signal as column vector (correctly plotted)
  ```
   >>> color = np.array([[1], [0], [0]])
   >>> mpl.colors.is_color_like(color)
   False
   >>> all(map(mpl.colors.is_color_like, color))
   False
   >>> pg.graphs.Sensor(3, k=1).plot(color)
   ```

I simplified in 3f4d4c4a2ef0ccc05063a4e06be8632622d4e295. Thanks for your contribution! :)
